libibverbs/device.c

```diff
@@ -262,6 +267,8 @@ struct ibv_context *__ibv_open_device(struct ibv_device *device)
 	context->cmd_fd = cmd_fd;
 	pthread_mutex_init(&context->mutex, NULL);
 
+	ibverbs_device_hold(device);
+
```

Comment:
If we want to address that in this series, we can go with a simple solution in the user area as follows:
As part of __ibv_open_device just *after* that the cmd_fd was opened successfully we can make sure that this is still the sysfs device that the application asked to open by re-reading the timestamp and comparing to what was already set on the ibv_device, In case that TS was changed it points on some "other" device and the API will return an error. Otherwise we can safely continue to allocate the context by calling the kernel device. In case that by that time the device was somehow dropped the API will fail and an error will be returned as expected. 

Does it make sense ?